File: Amazon.Lambda/Felipearpa.Tyche.PoolLayout/src/Application/GetOpenPoolLayouts.fs
namespace Felipearpa.Tyche.PoolLayout.Application

open Felipearpa.Tyche.PoolLayout.Domain

type GetOpenPoolLayouts(poolLayoutRepository: IPoolLayoutRepository) =
    member this.ExecuteAsync(maybeNext: string option) =
        poolLayoutRepository.GetOpenPoolLayoutsAsync(maybeNext)

File: Android/data/pool/src/main/java/com/felipearpa/tyche/data/pool/application/GetOpenPoolLayouts.kt
package com.felipearpa.tyche.data.pool.application

import com.felipearpa.tyche.data.pool.domain.PoolLayoutRepository

class GetOpenPoolLayouts internal constructor(private val poolLayoutRepository: PoolLayoutRepository) {
    suspend fun execute(next: String?, searchText: String?) =
        poolLayoutRepository.getOpenPoolLayouts(next = next, searchText = searchText)
}
